File: bookhub-back/src/main/java/com/example/bookhub_back/dto/employee/response/EmployeeSignUpApprovalsResponseDto.java
package com.example.bookhub_back.dto.employee.response;

import com.example.bookhub_back.common.enums.IsApproved;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class EmployeeSignUpApprovalsResponseDto {
    private Long approvalId;
    private Long employeeNumber;
    private String employeeName;
    private Long authorizerNumber;
    private String authorizerName;
    private String appliedAt;
    private IsApproved isApproved;
    private String deniedReason;
    private String updatedAt;
}

File: bookhub-front/src/dtos/employee/response/Employee-sign-up-approvals.response.dto.ts
import { ApprovalStatus } from "@/apis/enums/ApprovalType";

export interface EmployeeSignUpApprovalsResponseDto {
  approvalId: number;
  employeeNumber: number;
  employeeName: string;
  appliedAt: string;
  isApproved: ApprovalStatus;
  deniedReason: string | undefined;
  authorizerNumber: number;
  authorizerName: string;
  updatedAt: string;
}